### DISCUSSION TITLE
Sign-in form design updates

### DISCUSSION THREAD

**Discussion by bl-nero** [gravitational/teleport#41130] [Announcements]
## What happened?

Starting from version 15.2.2, we began updating the sign-in form design. As of version 15.2.5, here are some examples of how it may look like, depending on the Auth Service configuration:

<p align="center">
<img width="350" alt="Screenshot 2024-05-02 at 10 57 07" src="https://github.com/gravitational/teleport/assets/9391503/5802aadb-d3b4-41b1-9960-63dc0a709d82">
<img width="350" alt="Screenshot 2024-05-02 at 10 57 16" src="https://github.com/gravitational/teleport/assets/9391503/472c6e75-0d2b-4a13-9fa4-74eee1ddd153">
</p>

The most important change is that the secondary sign-in methods are now displayed right below the default one — with exception of a local username and password login, which still leads to a separate form if it's configured as a secondary method.

Between versions 15.2.2 and 15.2.4, the dialog was also significantly wider, and the SSO buttons were put in a two-column layout. We reverted this part of changes in version 15.2.5, going back to a more narrow design.

## Why did we change it?

The main motivation behind this change is promoting [passwordless authentication](https://goteleport.com/docs/access-controls/guides/passwordless/). Previously, even if passwordless authentication was enabled in the Auth Service configuration, it had to be configured as a [default method](https://goteleport.com/docs/access-controls/guides/passwordless/#optional-enable-passwordless-by-default) to be visible right away in the sign-in form; to access it, the user needed to click the "Other sign-in options" link. After this change, the passwordless method is always displayed on the first screen if available.

**Comment by webvictim** [comment]
Some related instructions to disable local or passwordless authentication if needed: https://github.com/gravitational/teleport/discussions/40996
